drivers/modem/ublox-sara-r4.c

```diff
@@ -647,9 +640,17 @@ static int pin_init(void)
 #if defined(DT_UBLOX_SARA_R4_0_MDM_VINT_GPIOS_CONTROLLER)
 	LOG_DBG("Waiting for MDM_VINT_PIN = 0");
 
-	do {
+	while (modem_pin_read(&mctx, MDM_VINT) != MDM_VINT_DISABLE) {
```

Comment:
what do you think about adding timeout test in order to be sure to exit loop ?
I had one case where I was always blocked in this loop.

Reply:
@gpaquet85 Can we hold off on that until after this is merged?  The loop doesn't change here, other than the values are checked at the top of the loop.  We can address it in a new PR with discussion.

Reply:
@mike-scott ok no problem. Indeed it can be adressed later